main.rb

```diff
@@ -0,0 +1,175 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y]
+      y += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y], y
+      y += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    access = []
+    my_each do |val|
+      access << val if yield val
+    end
+    access
+  end
+
+  def my_all?(par = nil)
+    return test_parm(par) unless par.nil?
+
+    return (my_all? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return false unless yield a
+    end
+    true
+  end
+
+  def test_parm(par)
+    if par.class == Class
+      my_all? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_all? { |x| x =~ par }
+    else
+      my_all? { |x| x == par }
+    end
+  end
+
+  def my_any?(par = nil)
+    return test_param_any(par) unless par.nil?
+
+    return (my_any? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return true if yield a
+    end
+    false
+  end
+
+  def test_param_any(par)
+    if par.class == Class
+      my_any? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_any? { |x| x =~ par }
+    else
+      my_any? { |x| x == par }
+    end
+  end
+
+  def my_none?(par = nil)
+    return test_par_none(par) unless par.nil?
+
+    return (my_none? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return false if yield a
+    end
+    true
+  end
+
+  def test_par_none(par)
+    if par.class == Class
+      my_none? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_none? { |x| x =~ par }
+    else
+      my_none? { |x| x == par }
+    end
+  end
+
+  def my_count(par = nil)
+    cnt = 0
+    return my_count { |x| x == par } unless par.nil?
+    return (my_count { |_x| true }) unless block_given?
+
+    my_each do |a|
+      cnt += 1 if yield a
+    end
+    cnt
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    new_array = []
+    my_each do |value|
+      new_array << (yield value)
+    end
+    new_array
+  end
```

Comment:
- [ ] Change your **my_map** so it takes a proc or a block, if both are given then you should only execute the proc as stated in step #12 of the [project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks)

```
EXAMPLE
a = [18, 22, 5, 6] 
my_proc = Proc.new {|num| num > 10 }
puts a.my_map(my_proc) {|num| num > 10 }

EXPECTED RESULT
true
true 
false
false
```